src/Circle.h

```diff
@@ -0,0 +1,251 @@
+/*
+ * LSST Data Management System
+ * Copyright 2014-2015 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_SG_CIRCLE_H_
+#define LSST_SG_CIRCLE_H_
+
+/// \file
+/// \brief This file declares a class for representing circular
+///        regions on the unit sphere.
+
+#include <iosfwd>
+
+#include "Region.h"
+#include "SpatialRelation.h"
+#include "UnitVector3d.h"
+
+
+namespace lsst {
+namespace sg {
+
+/// `Circle` is a circular region on the unit sphere that contains its
+/// boundary. Internally, the circle is represented by its center vector
+/// and the squared length of the chords between its center and points on
+/// its boundary. This yields a fast point-in-circle test but, unlike a
+/// representation that uses the center vector and cosine of the circle
+/// opening angle, remains accurate for circles with very small opening
+/// angles.
+class Circle : public Region {
+public:
+    static Circle empty() { return Circle(); }
+
+    static Circle full() { return Circle(UnitVector3d::Z(), 4.0); }
+
+    /// `squaredChordLengthFor` computes and returns the squared chord length
+    /// between points in S² that are separated by the given angle. The squared
+    /// chord length l² and angle θ are related by l² = 4 sin²(θ/2).
+    static double squaredChordLengthFor(Angle openingAngle);
+
+    /// `openingAngleFor` computes and returns the angular separation between
+    /// points in S² that are separated by the given squared chord length. The
+    /// squared chord length l² and angle θ are related by l² = 4 sin²(θ/2).
+    static Angle openingAngleFor(double squaredChordLength);
```

Comment:
It seems to me the returned angle here would be implicitly normalized -- was there a reason not to return the more strongly constrained NormalizedAngle here?